tests/pset.rs

```diff
@@ -0,0 +1,335 @@
+        let tx = bitcoind.client.get_raw_transaction(&txid, None).unwrap();
+        let tx_bytes = serialize(&tx);
+        let vout = tx
+            .output
+            .iter()
+            .position(|o| {
+                Address::from_script(&o.script_pubkey, bitcoin::Network::Regtest)
+                    .unwrap()
+                    .to_string()
+                    == pegin_address
+            })
+            .unwrap();
+
+        bitcoind
+            .client
+            .generate_to_address(101, &mainchain_address)
+            .unwrap();
+        let proof = bitcoind.client.get_tx_out_proof(&[txid], None).unwrap();
+        elementsd.generate(2);
+        let inputs = json!([ {"txid":txid, "vout": vout,"pegin_bitcoin_tx": tx_bytes.to_hex(), "pegin_txout_proof": proof.to_hex(), "pegin_claim_script": claim_script } ]);
+        let value = elementsd.call(
+            "createpsbt",
+            &[
+                inputs,
+                json!([
+                    {address_lbtc: "0.9", "blinder_index": 0},
+                    {"fee": "0.1" }
+                ]),
+                0.into(),
+                false.into(),
+            ],
+        );
+        let psbt_base64 = value.as_str().unwrap().to_string();
+        assert_eq!(elementsd.expected_next(&psbt_base64), "updater");
+        let psbt_base64 = elementsd.wallet_process_psbt(&psbt_base64);
+        assert_eq!(elementsd.expected_next(&psbt_base64), "extractor");
+
+        // TODO this fails because elements decodepsbt is not printing TxOut::asset (PSET_IN_WITNESS_UTXO)
+        //psbt_rtt(&elementsd, &psbt_base64);
+
+        let tx_hex = elementsd.finalize_psbt(&rtt(&psbt_base64));
+        assert!(elementsd.test_mempool_accept(&tx_hex));
+    }
+
+    fn rtt(base64: &str) -> String {
+        base64::encode(serialize(&psbt_from_base64(&base64)))
+    }
+
+    fn psbt_rtt(elementsd: &ElementsD, base64: &str) {
+        let a = elementsd.decode_psbt(&base64).unwrap();
+
+        let b_psbt = psbt_from_base64(&base64);
+        let mut b_bytes = serialize(&b_psbt);
+        let b_base64 = base64::encode(&b_bytes);
+        let b = elementsd.decode_psbt(&b_base64).unwrap();
+
+        assert_eq!(a, b);
+
+        let mut rng = rand::thread_rng();
+        let die = Uniform::from(0..b_bytes.len());
+        let mut tests = 0;
+        for _ in 0..1_000 {
+            let i = die.sample(&mut rng);
+            // ensuring decode prints all data inside psbt, randomly changing a byte,
+            // if the results is still decodable it should not be equal to initial value
+            b_bytes[i] = b_bytes[i].wrapping_add(1);
+            let base64 = base64::encode(&b_bytes);
+            if let Some(decoded) = elementsd.decode_psbt(&base64) {
+                assert_ne!(a, decoded, "{} with changed byte {}", b_bytes.to_hex(), i);
+                tests += 1;
+            }
+            b_bytes[i] = b_bytes[i].wrapping_sub(1);
+        }
+        assert!(tests > 0)
```

Comment:
Are you betting that at least one in a thousand a changed byte results in a valid PSET? :sweat_smile: Not sure why you'd want to enforce that? You could also just loop for anything between "both at least 1000 times and at least one test performed" or so?

Reply:
lol, I added during dev to ensure I was passing in the if branch and forgot to remove, removed in 682eb8a5bc972e28ccea8997ec582aa2d28769a1